kratos/containers/csr_matrix.h

```diff
@@ -677,8 +693,71 @@ class CsrMatrix final
         }
     }
 
-    //TODO
-    //NormFrobenius
+    double NormDiagonal() const
+    {
+        //TODO: Investigate why do we need the template keyword below
+        const double diagonal_norm = IndexPartition<IndexType>(size1()).template for_each<SumReduction<double>>([&](IndexType Index) {
+            const IndexType row_begin = index1_data()[Index];
+            const IndexType row_end = index1_data()[Index+1];
+            for (IndexType j = row_begin; j < row_end; ++j) {
+                if (index2_data()[j] == Index) {
+                    return std::pow(value_data()[j], 2);
+                }
+            }
+            return 0.0;
+        });
+
+        return std::sqrt(diagonal_norm);
+    }
+
+    double MaxDiagonal() const
+    {
+        //TODO: Investigate why do we need the template keyword below
+        return IndexPartition<IndexType>(size1()).template for_each<MaxReduction<double>>([&](IndexType Index) {
+            const IndexType row_begin = index1_data()[Index];
+            const IndexType row_end = index1_data()[Index+1];
+            for (IndexType j = row_begin; j < row_end; ++j) {
+                if (index2_data()[j] == Index) {
+                    return std::abs(value_data()[j]);
+                }
+            }
+            return std::numeric_limits<double>::lowest();
+        });
+    }
+
+    double MinDiagonal() const
+    {
+        //TODO: Investigate why do we need the template keyword below
+        return IndexPartition<IndexType>(size1()).template for_each<MinReduction<double>>([&](IndexType Index) {
+            const IndexType row_begin = index1_data()[Index];
+            const IndexType row_end = index1_data()[Index+1];
+            for (IndexType j = row_begin; j < row_end; ++j) {
+                if (index2_data()[j] == Index) {
+                    return std::abs(value_data()[j]);
+                }
+            }
+            return std::numeric_limits<double>::max();
+        });
+    }
+
+    SizeType GraphDegree(const IndexType I) const
```

Comment:
here we need a comment describing what it does

Reply:
Same comment. It is only used in the skyline LU solver (we will directly implement it in there).